app/models/Service.scala

```diff
@@ -64,9 +96,216 @@ object Service {
     }
   }
 
+
+  /**
+    * Identity key for CSV file
+    */
+  val csvKey = "service";
+
+  /**
+    * Header of csvFile. Defines the column name and order.
+    */
+  val csvTitle = List(
+    csvKey, "id", "description", "environmentName", "typeRequest", "httpMethod",
+    "localTarget", "remoteTarget", "useMockGroup", "mockGroupId",
+    "timeoutms", "recordContentData", "recordData"
+  )
+
+  /**
+    * Get all services, csv format.
+    *
+    * @return List of services, csv format
+    */
+  def fetchCsv(): List[String] = {
+    val f = findAll.map(service => service.map(s => s.toCSV()))
+    Await result(f, 5.seconds)
+  }
+
+  /**
+    * Get the header as a csv string
+    *
+    * @return
+    */
+  def fetchCsvHeader(): String = {
+    "#" + Service.csvTitle.mkString(",")
+  }
+
+  /**
+    * Upload a csvLine => insert service & potential environment.
+    *
+    * @param csvLine line in csv file
+    * @return nothing
+    */
+  def uploadCSV(csvLine: String): Either[ErrorUploadCsv, Boolean] = {
+
+    val dataCsv = csvLine.split(",")
+
+    if (dataCsv.size != csvTitle.size)
+      throw new Exception("Please check csvFile, " + csvTitle.size + " fields required")
+
+    if (dataCsv(0) == csvKey) {
+      val uploadFuture = uploadEnvironment(dataCsv).flatMap {
+        uploaded => uploadService(dataCsv)
+      }
+      Right(Await.result(uploadFuture, 10.seconds))
+    } else {
+      Left(ErrorUploadCsv(s"First column ${dataCsv(0)} is not recognized as ${csvKey} "))
+    }
+
+  }
+
+  /**
+    * Upload a new Environment from the service
+    *
+    * @param dataCsv
+    */
+  private def uploadEnvironment(dataCsv: Array[String]): Future[Boolean] = {
+
+    val envS = dataCsv(csvTitle.indexOf("environmentName")).trim
+    val localTarget = dataCsv(csvTitle.indexOf("localTarget"))
+    if (envS.isEmpty) {
+      throw new Exception(s"environmentName is mandatory when uploading the service ${localTarget}.")
+    }
+
+    def insertEnvironment(env: Environment, localTarget: String): Future[Boolean] = {
+      val u = Environment.insert(env)
+      u.map {
+        case res => {
+          Logger.info(s"Created new default environment ${env.name} for service ${localTarget}")
+          true
+        }
+      }
+    }
+
+    // Search the environment by its name
+    val potentialEnvironmentF = Environment.findByName(envS, cached = false)
+    potentialEnvironmentF.flatMap {
+      case Some(e) => {
+        // Environment exists, so we do nothing because we don't have any other info to update
+        Logger.info(s"Environment ${e.name} exists for service ${localTarget}")
+        Future.successful(true)
+      }
+      case None => {
+        // Create a default environment with name
+        Logger.info(s"Environment ${envS} does not exist for service ${localTarget}")
+        val env = new Environment(
+          None,
+          envS,
+          List()
+        )
```

Comment:
No `new` for case classes + named parameters to understand what `None`and `List()` actually are